packages/components/src/components/progressbar/progressbar.component.ts

```diff
@@ -0,0 +1,143 @@
+import { CSSResult, html, nothing } from 'lit';
+import { property } from 'lit/decorators.js';
+import styles from './progressbar.styles';
+import FormfieldWrapper from '../formfieldwrapper';
+import { DEFAULTS, VARIANT } from './progressbar.constants';
+import { Variant } from './progressbar.types';
+import { DataAriaLabelMixin } from '../../utils/mixins/DataAriaLabelMixin';
+import { VALIDATION } from '../formfieldwrapper/formfieldwrapper.constants';
+
+/**
+ * mdc-progressbar component visually represents a progress indicator, typically used to show
+ * the completion state of an ongoing process (e.g., loading, file upload, etc.).
+ * It contains an optional label and an optional helper text.
+ *
+ * - It supports mainly two types: Default and Inline
+ * - It supports three validation variants: Default, Success and Error.
+ *
+ * This component is created by extending FormfieldWrapper.
+ *
+ * @tagname mdc-progressbar
+ *
+ * @dependency mdc-icon
+ * @dependency mdc-text
+ *
+ * @cssproperty --mdc-progressbar-default-background-color - Background color of the progressbar when inactive.
+ * @cssproperty --mdc-progressbar-default-active-background-color - Background color of the progressbar when active.
+ * @cssproperty --mdc-progressbar-success-background-color - Background color of the progressbar when in success state.
+ * @cssproperty --mdc-progressbar-error-background-color - Background color of the progressbar when in error state.
+ * @cssproperty --mdc-progressbar-height - The height of the progressbar.
+ * @cssproperty --mdc-progressbar-border-radius - The border radius of the progressbar.
+ * @cssproperty --mdc-progressbar-label-color - Color of the progressbar label text.
+ * @cssproperty --mdc-progressbar-label-lineheight - Line height of the label text.
+ * @cssproperty --mdc-progressbar-label-fontsize - Font size of the label text.
+ * @cssproperty --mdc-progressbar-label-fontweight - Font weight of the label text.
+ * @cssproperty --mdc-progressbar-help-text-color - Color of the help text.
+ */
+class Progressbar extends DataAriaLabelMixin(FormfieldWrapper) {
+  /**
+   * Types of the progressbar
+   * - **Default**
+   * - **Inline**
+   *
+   * @default default
+   */
+  @property({ type: String, reflect: true })
+  variant: Variant = DEFAULTS.VARIANT;
+
+  /**
+   * The current progress as a percentage, 0 to 100.
+   * The value will be clamped between 0 and 100.
+   * @default 0
+   */
+  @property({ type: String, reflect: true })
+  value: string = '0';
+
+  /**
+   * Define error state of the progressbar
+   * - **true**
+   * - **false**
+   * @default false
+   */
+  @property({ type: Boolean, attribute: 'error' }) error = false;
+
+  constructor() {
+    super();
+    this.id = '';
+    this.disabled = undefined as unknown as boolean;
+  }
+
+  /**
+   * Ensures that the value is clamped between 0 and 100
+   * @returns The clamped value
```

Comment:
```suggestion
   * @returns The clamped value
   * @internal
```

Adding this would not show the property in storybook.